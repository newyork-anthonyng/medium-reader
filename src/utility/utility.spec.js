/* global describe, it, expect */
const myModule = require('./utility')
const {
  verifyMediumLink
} = myModule

describe('verifyMediumLink', () => {
  it('should return link if valid', () => {
    const link = 'https://medium.com'

    expect(verifyMediumLink(link)).toEqual(link)
  })

  it('should prepend with https', () => {
    const invalidLink = 'medium.com'

    expect(verifyMediumLink(invalidLink)).toMatchSnapshot()
  })

  it('should work with malformed links', () => {
    const invalidLink = 'uhohsomethinghappenedmedium.com/foobar'

    expect(verifyMediumLink(invalidLink)).toMatchSnapshot()
  })
})
