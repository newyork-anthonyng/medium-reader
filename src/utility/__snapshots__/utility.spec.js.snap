// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verifyMediumLink should prepend with https 1`] = `"https://medium.com"`;

exports[`verifyMediumLink should work with malformed links 1`] = `"https://medium.com/foobar"`;
